client/components/data/domain-management/email/index.jsx

```diff
@@ -48,84 +51,125 @@ function getStateFromStores( props ) {
 	};
 }
 
-const EmailData = React.createClass( {
-	displayName: 'EmailData',
+export class EmailData extends Component {
+	static propTypes = {
+		context: PropTypes.object.isRequired,
+		fetchSitePlans: PropTypes.func.isRequired,
+		products: PropTypes.object.isRequired,
+		selectedDomainName: PropTypes.string,
+		selectedSite: PropTypes.object.isRequired,
+		sitePlans: PropTypes.object.isRequired,
+		googleAppsUsers: PropTypes.array.isRequired,
+		googleAppsUsersLoaded: PropTypes.bool.isRequired,
+	};
 
-	propTypes: {
-		component: React.PropTypes.func.isRequired,
-		context: React.PropTypes.object.isRequired,
-		productsList: React.PropTypes.object.isRequired,
-		selectedDomainName: React.PropTypes.string,
-		sitePlans: React.PropTypes.object.isRequired,
-		sites: React.PropTypes.object.isRequired,
-		googleAppsUsers: React.PropTypes.array.isRequired,
-		googleAppsUsersLoaded: React.PropTypes.bool.isRequired
-	},
+	constructor( props ) {
+		super( props );
 
-	mixins: [ observe( 'productsList' ) ],
+		props.loadDomainsAndSitePlans();
```

Comment:
Let's slim this down to just `fetchDomains` since we have a query component for site plans. We can then remove some of the manual plan dispatches below.
```
import QuerySitePlans from 'components/data/query-site-plans';
<QuerySitePlans siteId={ siteId } />
```